lib/grovepi/pivotpi/PCA9686.ex

```diff
@@ -0,0 +1,78 @@
+defmodule GrovePi.PivotPi.PCA9685 do
+  @moduledoc false
+
+  # References
+  # https://github.com/DexterInd/PivotPi/tree/master/Software/Python
+  # https://www.nxp.com/docs/en/data-sheet/PCA9685.pdf
+
+  # registers/etc:
+  @pca9685_address     0x40
+  @mode1               0x00
+  @mode2               0x01
+  # @subadr1             0x02 # Unused
+  # @subadr2             0x03 # Unused
+  # @subadr3             0x04 # Unused
+  @prescale            0xfe
+  @led0_on_l           0x06
+  @led0_on_h           0x07
+  @led0_off_l          0x08
+  @led0_off_h          0x09
+  @all_led_on_l        0xfa
+  @all_led_on_h        0xfb
+  @all_led_off_l       0xfc
+  @all_led_off_h       0xfd
+
+  # bits:
+  @allcall             0x01
+  @outdrv              0x04
+  @sleep               0x10
+  @invrt               0x10
+  # @restart             0x80 # Unused
+
+  @default_freq        60
+  @default_mode1       0x01
+
+  alias GrovePi.{Board}
+  import Bitwise
+
+  def start() do
+    set_all_pwm(0, 0)
+    set_initial_config()
+    set_pwm_freq(@default_freq)
+  end
+
+  defp set_initial_config() do
+    send_cmd(<<@mode2, (@outdrv ||| @invrt)>>) # Totem pole drive, and inverted signal.
+    send_cmd(<<@mode1, @allcall>>)
+    Process.sleep(50)
+    send_cmd(<<@mode1, @default_mode1>>)
+    Process.sleep(50)
+  end
+
+  defp set_pwm_freq(freq_hz) do
+    prescale_val = round(((25000000.0 / 4096.0) / freq_hz) - 1.0)
+    sleep_mode = @default_mode1 ||| @sleep
+    send_cmd(<<@mode1, sleep_mode>>)
+    send_cmd(<<@prescale, prescale_val>>)
+    send_cmd(<<@mode1, @default_mode1>>)
+    Process.sleep(50)
+  end
+
+  def set_pwm(channel, on, off) do
+    send_cmd(<<@led0_on_l+4*channel, on &&& 0xFF>>)
+    send_cmd(<<@led0_on_h+4*channel, on >>> 8>>)
+    send_cmd(<<@led0_off_l+4*channel, off &&& 0xFF>>)
+    send_cmd(<<@led0_off_h+4*channel, off >>> 8>>)
```

Comment:
All the commands seem very magical here. Is there some intention revealing name that we can give to the led channel formula? There is also has a lot of repetition with `set_all_pwm` it would be nice if we could pull out a function that handles that logic.

Reply:
I pulled the calculations within these functions out into their own functions.